x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mappings_logic.ts

```diff
@@ -0,0 +1,337 @@
+        handleAttributeSelectorChange: (_, { value }) => value,
+        resetState: () => 'username',
+      },
+    ],
+    selectedEngines: [
+      new Set(),
+      {
+        setRoleMappingData: (_, { roleMapping }) =>
+          roleMapping ? new Set(roleMapping.engines.map((engine) => engine.name)) : new Set(),
+        handleAccessAllEnginesChange: () => new Set(),
+        handleEngineSelectionChange: (engines, { engineName, selected }) => {
+          const newSelectedEngineNames = new Set(engines as Set<string>);
+          if (selected) {
+            newSelectedEngineNames.add(engineName);
+          } else {
+            newSelectedEngineNames.delete(engineName);
+          }
+
+          return newSelectedEngineNames;
+        },
+      },
+    ],
+    availableAuthProviders: [
+      [],
+      {
+        setRoleMappingData: (_, { authProviders }) => authProviders,
+      },
+    ],
+    selectedAuthProviders: [
+      [ANY_AUTH_PROVIDER],
+      {
+        handleAuthProviderChange: (previous, { value }) => {
+          const previouslyContainedAny = previous.includes(ANY_AUTH_PROVIDER);
+          const newSelectionsContainAny = value.includes(ANY_AUTH_PROVIDER);
+
+          if (value.length < 1) return [ANY_AUTH_PROVIDER];
+          if (value.length === 1) return value;
+          if (!newSelectionsContainAny) return value;
+          if (previouslyContainedAny) return value.filter((v) => v !== ANY_AUTH_PROVIDER);
+          return [ANY_AUTH_PROVIDER];
+        },
+        setRoleMappingData: (_, { roleMapping }) =>
+          roleMapping ? roleMapping.authProvider : [ANY_AUTH_PROVIDER],
+      },
+    ],
+  },
+  listeners: ({ actions, values }) => ({
+    initializeRoleMappings: async () => {
+      const { http } = HttpLogic.values;
+      const route = '/api/app_search/role_mappings';
+
+      try {
+        const response = await http.get(route);
+        actions.setRoleMappingsData(response);
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+    },
+    initializeRoleMapping: async ({ roleId }) => {
+      const { http } = HttpLogic.values;
+      const { navigateToUrl } = KibanaLogic.values;
+      const route = roleId
+        ? `/api/app_search/role_mappings/${roleId}`
+        : '/api/app_search/role_mappings/new';
+
+      try {
+        const response = await http.get(route);
+        actions.setRoleMappingData(response);
+      } catch (e) {
+        navigateToUrl(ROLE_MAPPINGS_PATH);
+        flashAPIErrors(e);
+      }
+    },
+    handleDeleteMapping: async () => {
+      const { roleMapping } = values;
+      if (!roleMapping) return;
+
+      const { http } = HttpLogic.values;
+      const { navigateToUrl } = KibanaLogic.values;
+      const route = `/api/app_search/role_mappings/${roleMapping.id}`;
+
+      if (window.confirm(DELETE_ROLE_MAPPINGS_MESSAGE)) {
+        try {
+          await http.delete(route);
+          navigateToUrl(ROLE_MAPPINGS_PATH);
```

Comment:
Should we set a queued success message here before navigating to the role mappings page?

Reply:
Agreed. Nice catch!